### PR TITLE
feat(core): add support for requirements and patch.diff

### PR DESCRIPTION
**Pull request by kptkin**
Description
-----------

- Fixes WB-NNNNN
- Fixes #NNNN

What does the PR do?

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 826940d</samp>

Add a feature to save and sync the pip packages used in the environment to a `requirements.txt` file. Use the `pkg_resources` module and the `_save_requirements` setting in `wandb.init`.

Testing
-------
How was this PR tested?

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 826940d</samp>

> _`_save_requirements`_
> _Capture pip packages in file_
> _Autumn leaves of code_

### CODE REVIEW

**Review by dmitryduev — APPROVED** [review]

File: wandb/sdk/internal/system/system_info.py [hunk 1]
```diff
@@ -245,10 +200,35 @@ def probe(self) -> Dict[str, Any]:
...
+                dict(files=[(CONDA_ENVIRONMENTS_FNAME, "now")])
+            )
+        except Exception as e:
+            logger.exception(f"Error saving conda packages: {e}")
+        logger.debug("Saving conda packages done")
+
     def publish(self, system_info: dict) -> None:
         # save pip, conda, code patches to disk
         if self.settings._save_requirements:
-            self._save_pip()
```
Comment (dmitryduev) [inline wandb/sdk/internal/system/system_info.py:251]: so the next python release will miss it? prob should put it back for now and check if it's running against core

**Review by dmitryduev — APPROVED** [review]
Excellente! 🐧

File: core/pkg/server/handler.go [hunk 2]
```diff
@@ -546,7 +566,71 @@ func (h *Handler) handleCodeSave() {
...
+		If there are submodules, you can use the --submodule=diff option to make git diff recurse into them:
+		git diff HEAD --submodule=diff
+
+		To check if there are submodules:
+		git submodule status
+		(should return nothing if there are no submodules)
+
+		Get diff of current working tree vs last commit on upstream branch
+		git diff @{u}
+	*/
```
Comment (kptkin) [inline core/pkg/server/handler.go:587]:
```suggestion
	// capture git state
```

File: wandb/sdk/internal/system/system_info.py [hunk 3]
```diff
@@ -245,10 +200,31 @@ def probe(self) -> Dict[str, Any]:
...
+            with open(
+                os.path.join(self.settings.files_dir, CONDA_ENVIRONMENTS_FNAME), "w"
+            ) as f:
+                subprocess.call(
+                    ["conda", "env", "export"], stdout=f, stderr=subprocess.DEVNULL
+                )
+            # self.backend.interface.publish_files(
+            #     dict(files=[(CONDA_ENVIRONMENTS_FNAME, "now")])
+            # )
+        except Exception as e:
```
Comment (kptkin) [inline wandb/sdk/internal/system/system_info.py:221]:
```suggestion
        except Exception as e:
```

**Pull request merged**